components/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/impl/InterceptingIdentityStore.java

```diff
@@ -331,6 +331,95 @@ public User execute() throws IdentityStoreException, UserNotFoundException {
     }
 
     @Override
+    public List<User> listUsers(List<Claim> claims, int offset, int length) throws IdentityStoreException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(IdentityStoreConstants.CLAIM_LIST, claims);
+        eventProperties.put(IdentityStoreConstants.OFFSET, offset);
+        eventProperties.put(IdentityStoreConstants.LENGTH, length);
+
+        Event event = new Event(IdentityStoreInterceptorConstants.PRE_LIST_USERS_BY_CLAIMS, eventProperties);
+        IdentityMgtMessageContext messageContext = new IdentityMgtMessageContext(event);
+
+        try {
+            eventService.handleEvent(messageContext);
+        } catch (EventException e) {
+
+            String message = String.format("Error while handling %s event.", IdentityStoreInterceptorConstants
+                    .PRE_LIST_USERS_BY_CLAIMS);
+            throw new IdentityStoreException(message, e);
+        }
+
+        List<User> users = identityStore.listUsers(claims, offset, length);
+
+        eventProperties = new HashMap<>();
+        eventProperties.put(IdentityStoreConstants.CLAIM_LIST, claims);
+        eventProperties.put(IdentityStoreConstants.OFFSET, offset);
+        eventProperties.put(IdentityStoreConstants.LENGTH, length);
+        eventProperties.put(IdentityStoreConstants.USER_LIST, users);
+
+        event = new Event(IdentityStoreInterceptorConstants.POST_LIST_USERS_BY_CLAIMS, eventProperties);
+        messageContext = new IdentityMgtMessageContext(event);
+
+        try {
+            eventService.handleEvent(messageContext);
+        } catch (EventException e) {
+
+            String message = String.format("Error while handling %s event.", IdentityStoreInterceptorConstants
+                    .POST_LIST_USERS_BY_CLAIM);
+            throw new IdentityStoreException(message, e);
+        }
+
+        return users;
+    }
+
+    @Override
+    public List<User> listUsers(List<Claim> claims, int offset, int length, String domainName)
+            throws IdentityStoreException {
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put(IdentityStoreConstants.CLAIM_LIST, claims);
+        eventProperties.put(IdentityStoreConstants.OFFSET, offset);
+        eventProperties.put(IdentityStoreConstants.LENGTH, length);
+        eventProperties.put(IdentityStoreConstants.DOMAIN_NAME, domainName);
+
+        Event event = new Event(IdentityStoreInterceptorConstants.PRE_LIST_USERS_BY_CLAIMS_DOMAIN, eventProperties);
+        IdentityMgtMessageContext messageContext = new IdentityMgtMessageContext(event);
+
+        try {
+            eventService.handleEvent(messageContext);
```

Comment:
EventService no longer supports `handleEvent`. For the time being return `identityStore.listUsers` and add a TODO. We can implement the event triggering part later.